sycl/doc/design/CompileTimeProperties.md

```diff
@@ -110,33 +110,56 @@ template </* ... */> class
 
 } // namespace sycl::ext::oneapi
 ```
+An entry is added to the `@llvm.global.annotations` list, as described in
+[IR representation via `@llvm.global.annotations`][5], whenever a 
+[LLVM IR global variable][3] which is associated with a type decorated with
+`[[__sycl_detail__::add_ir_annotations_global_variable()]]` attribute is 
+defined. Associations may include, but are not limited to:
 
-The device compiler only uses the
-`[[__sycl_detail__::add_ir_attributes_global_variable()]]` attribute when the
-decorated type is used to create an [LLVM IR global variable][3] and the global
-variable's type is either:
+* A type that is directly decorated by the attribute
+* An array of the type that is decorated by the attribute
+* A type whose member variable is decorated by the attribute
 
-* The type that is decorated by the attribute, or
-* An array of the type that is decorated by the attribute.
+In the example below, the annotated type `fpga_mem` itself has
+`[[__sycl_detail__::add_ir_annotations_global_variable()]]` attribute. Both the
+attributes on the `device_global` and on `fpga_mem` will generate entries in the
+`@llvm.global.annotations` list.
 
-[3]: <https://llvm.org/docs/LangRef.html#global-variables>
+```cpp
+namespace sycl::ext::intel {
+
+template <typename T, typename ...Props>
+class
+#ifdef __SYCL_DEVICE_ONLY__
+  [[__sycl_detail__::add_ir_annotations_global_variable(
+    "sycl-resource", Props::meta_name..., "DEFAULT", Props::meta_value...
+    )]]
+#endif
+  fpga_mem<T, properties<Props...>> {/*...*/};
 
-The device compiler front-end silently ignores the attribute when the decorated
-type is used in any other way.
+} // namespace sycl::ext::intel
 
-When the device compiler front-end creates a global variable from the decorated
-type as described above, it also adds one IR attribute to the global variable
-for each property using
-[`GlobalVariable::addAttribute(StringRef, StringRef)`][4].  If the property
-value is not already a string, it converts it to a string as described in
-[IR representation as IR attributes][5].
+namespace intel = sycl::ext::intel;
+namespace oneapi = sycl::ext::oneapi;
 
-[4]: <https://llvm.org/doxygen/classllvm_1_1GlobalVariable.html#a6cee3c634aa5de8c51e6eaa4e41898bc>
-[5]: <#ir-representation-as-ir-attributes>
+oneapi::device_global<
+  intel::fpga_mem<
+    int[4],
+    decltype(oneapi::properties(intel::word_size<2>))>,
+  decltype(oneapi::properties(oneapi::host_access_none))> dg;
+```
+
+The device compiler front-end ignores the
+`[[__sycl_detail__::add_ir_annotations_global_variable()]]` attribute when it is
+not applied to the object definition, or a variable of a type decorated with 
+this attribute is not declared at namespace scope. 
```

Comment:
```suggestion
not applied to an object definition, or a variable of a type decorated with 
this attribute is not declared at namespace scope. 
```

Also the wording of the second half is imprecise as it doesn't say what happens if there are some variables of this decorated type declared at namespace scope and other declared at other scopes.  I think what you want to say is that if an object of a decorated type is not declared at namespace scope then the decoration is ignored _for that object_.